radix-engine/src/model/transaction_processor/executables.rs

```diff
@@ -433,9 +443,24 @@ impl<'a> Executor for TransactionProcessorRunInvocation<'a> {
                     metadata,
                     access_rules,
                 }) => {
+                    let code = blobs_by_hash
+                        .get(&code.0)
+                        .ok_or(RuntimeError::ApplicationError(
+                            ApplicationError::TransactionProcessorError(
+                                TransactionProcessorError::BlobNotFound(code.clone()),
+                            ),
+                        ))?;
+                    let abi = blobs_by_hash
+                        .get(&abi.0)
+                        .ok_or(RuntimeError::ApplicationError(
+                            ApplicationError::TransactionProcessorError(
+                                TransactionProcessorError::BlobNotFound(abi.clone()),
+                            ),
+                        ))?;
+                    // TODO: remove clone by allowing invocation to have references, like in TransactionProcessorRunInvocation.
```

Comment:
I imagine this should be pretty trivial to change in this PR to save adding a TODO? 👍 

Also - I'm surprised we need a double clone here, but I might be missing something! Can we avoid one with a deref?

Reply:
> I imagine this should be pretty trivial to change in this PR to save adding a TODO? 👍

Ahh maybe not. I guess we need to then include lifetime information up the whole tree of instructions etc...
Okay, ignore this :D

Reply:
Haha, I tried to update. 

The whole invocation enum hierachiy needs to be updated if we're not separating "transient" invocation and seralized invocation. 

There is very little performance loss from my benchmark, so I didn't pursue it for now.